### ISSUE TITLE
Add support for identifying third party commands

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#53192]
As plugins start adding commands using the the [Command Palette API](https://make.wordpress.org/core/2023/07/17/introducing-the-wordpress-command-palette-api/), it would be nice if each command registered had the associated plugin name visible in the command. This way if there were competing commands (either between third-parties, or between a third-party and core) you would be able to identify which command you were looking for. 

Not every command would utilize the plugin's branded icon (perhaps there isn't one even)—so we can't rely on just the visual. 

It could be interesting if the plugin name was used as a fallback, but the command could register an alternate. There are considerations on both fronts, but I'm concerned about plugins that use extensive names (i.e. "Ninja Forms Contact Form – The Drag and Drop Form Builder for WordPress" — where the visible name should be "Ninja Forms". 

Related to https://github.com/WordPress/gutenberg/issues/53190 in theory. Perhaps the identification here—and type/category within that concept—are the same idea. 

### Visual 

<img width="620" alt="third-party-commands" src="https://github.com/WordPress/gutenberg/assets/1813435/7b86bc68-ea7c-4acc-aedd-c3c12d24bae2">

**Comment by jasmussen** [comment]
Good and valid issue. The visual looks good too, but I'm wondering, what can we do to reduce the visual footprint so as to make each item more legible? The more complex the silhouette, the longer it will take to parse, in my experience. 

In this case, including both icons and the textual representation seems a lot, and text-wise there's a concern of very long plugin names. Icon-wise, what if a plugin does not offer an icon?

Would be possible to add a smaller icon only for plugin additions, like we have a "lock-small" icon? Or perhaps when both core and plugin commands are found, they are grouped separately with core commands on the top, and then a group for plugin commands below? (I know we just removed a grouping border, but perhaps it can be a different type of separator or just whitespace.) An additional option might be to show a plugin name only on highlight/focus, and then make sure it's elided if the name is longer than n characters (and denoting max-chars in the API).

**Comment by richtabor** [comment]
> Icon-wise, what if a plugin does not offer an icon?

My thought is that icons are entirely command-based. 

Third-party plugins can either add their own brand icon (like in the Jetpack "Activity log" command above), a relative icon (like in the Yoast "Social previews" command, or no icon at all per https://github.com/WordPress/gutenberg/issues/53193.

The plugin/category/source detail on the right would be consistent though. 

> Would be possible to add a smaller icon only for plugin additions, like we have a "lock-small" icon? Or perhaps when both core and plugin commands are found, they are grouped separately with core commands on the top, and then a group for plugin commands below?

I don't think we should separate core and plugin commands. Perhaps if there's a 1:1 match, core's should be prioritized, but generally I think they could be intermixed just fine. 

> An additional option might be to show a plugin name only on highlight/focus, and then make sure it's elided if the name is longer than n characters (and denoting max-chars in the API).

The bit there is that I wouldn't want to require navigating to a command to understand what it is. Imagine "General settings" commands, but one for core, one for Jetpack, another for Yoast, etc — but the palette reads "General settings", "General settings", "General settings". 

Yes, we should truncate the category/type/source text if we roll it out.

**Comment by richtabor** [comment]
An alternative, but we already decided against this pattern. 

Plus, I think it's more difficult to parse and doesn't help provide additional context that could be employed as https://github.com/WordPress/gutenberg/issues/53190#issuecomment-1658956139 (if "Post" was set as the category/type/source on the right). 

<img width="620" alt="third-party-commands" src="https://github.com/WordPress/gutenberg/assets/1813435/ee77669f-e175-48e4-80e4-2261312fbd0a">

**Comment by jasmussen** [comment]
> The bit there is that I wouldn't want to require navigating to a command to understand what it is. Imagine "General settings" commands, but one for core, one for Jetpack, another for Yoast, etc — but the palette reads "General settings", "General settings", "General settings".

Right, definitely don't want mystery meat navigation, but the idea here would work only if the note on the right was the _plugin short-name_, and not a _category of action_.

**Comment by Vrishabhsk** [comment]
Hi @richtabor 👋 
- I have raised a PR for the first iteration for implementing this : https://github.com/WordPress/gutenberg/pull/68573
- I have gone ahead with the utilization of the `Plugin Name` with `12 character limit`

Let me know your thoughts on this. Thanks
